Google.Api.Gax.Grpc/CallSettings.cs

```diff
@@ -215,6 +262,19 @@ public static CallSettings FromHeader(string name, string value)
             return FromHeaderMutation(metadata => metadata.Add(name, value));
         }
 
+        /// <summary>
+        /// Creates a <see cref="CallSettings"/> that will replace all the values associated
+        /// to the header of <paramref name="name"/>, if present, or add it if not.
+        /// </summary>
+        /// <param name="name">The name of the header to replace. Must not be null.</param>
+        /// <param name="value">The new value that should be associated to the header. Must not be null.</param>
+        /// <returns>A new instance.</returns>
+        public static CallSettings FromHeaderReplacement(string name, string value)
+        {
+            GaxPreconditions.CheckNotNull(value, nameof(value));
+            return FromHeaderReplacement(name, _ => new string[] { value });
```

Comment:
This will create an array on every usage. Maybe create the array once and then capture it? We know that `FromHeaderReplacement` won't modify it.